tests/structured_conf/test_structured_config.py

```diff
@@ -864,6 +864,135 @@ def test_create_untyped_list(self, class_type: str) -> None:
         assert cfg.list == [1, 2]
         assert cfg.opt_list is None
 
+    class TestInstantiateStructuredConfigs:
+        @pytest.fixture
+        def module(self, class_type: str) -> Any:
+            module: Any = import_module(class_type)
+            return module
+
+        def round_trip_to_object(self, input_data: Any) -> Any:
+            serialized = OmegaConf.create(input_data)
+            round_tripped = OmegaConf.to_object(serialized)
+            return round_tripped
+
+        def test_basic(self, module: Any) -> None:
+            user = self.round_trip_to_object(module.User("Bond", 7))
+            assert isinstance(user, module.User)
+            assert type(user) is module.User
+            assert user.name == "Bond"
+            assert user.age == 7
+
+        def test_basic_with_missing(self, module: Any) -> None:
+            user = self.round_trip_to_object(module.User())
+            assert isinstance(user, module.User)
+            assert type(user) is module.User
+            assert user.name == MISSING
+            assert user.age == MISSING
```

Comment:
I think instantiation should fail if there are MISSING fields that should be used to instantiate objects.
Is this the motivation for the `to_container throw_if_missing` PR?

Reply:
> Is this the motivation for the `to_container throw_if_missing` PR?

Yes, exactly.

Reply:
Instantiation now fails of there are MISSING fields.